core/src/test/java/com/linecorp/armeria/server/AnnotatedHttpServiceTest.java

```diff
@@ -482,6 +485,20 @@ public String post() {
         }
     }
 
+    @Converter(target = String.class, value = UnformattedStringConverter.class)
+    public static class MyAnnotatedService10 {
+
+        @Get("/syncThrow")
+        public String sync() {
+            throw new IllegalArgumentException("foo");
+        }
+
+        @Get("/asyncThrow")
+        public CompletableFuture<String> async() {
+            throw new IllegalArgumentException("bar");
```

Comment:
What about CompletableFuture.completedExceptionally?

Reply:
ah.. but then I need to think make it even in `HttpResponse.from(CompletionStage<? extends HttpResponse> stage)` should behave like this too.

Reply:
How about adding another method that returns a failed `CompletableFuture` for completeness then?